metron-stellar/stellar-common/src/main/java/org/apache/metron/stellar/common/shell/DefaultStellarShellExecutor.java

```diff
@@ -0,0 +1,398 @@
+
+  @Override
+  public Map<String, VariableResult> getState() {
+    return UnmodifiableMap.decorate(variables);
+  }
+
+  public Map<String, Object> getVariables() {
+    return Maps.transformValues(variables, (v) -> v.getResult());
+  }
+
+  @Override
+  public Context getContext() {
+    return context;
+  }
+
+  /**
+   * Creates a Zookeeper client.
+   * @param zookeeperUrl The Zookeeper URL.
+   */
+  private Optional<CuratorFramework> createZookeeperClient(Optional<String> zookeeperUrl) {
+    Optional<CuratorFramework> client = Optional.empty();
+
+    // can only create client, if have valid zookeeper URL
+    if(zookeeperUrl.isPresent()) {
+      String url = zookeeperUrl.get();
+      if (StringUtils.isNotBlank(url)) {
+
+        LOG.debug(String.format("Connecting to Zookeeper; url=%s", url));
+        CuratorFramework c = ConfigurationsUtils.getClient(url);
+        c.start();
+        client = Optional.of(c);
+      }
+    }
+
+    return client;
+  }
+
+  /**
+   * Creates a Context initialized with configuration stored in Zookeeper.
+   * @param properties Properties to configure the context.
+   * @param zkClient An optional Zookeeper client.
+   */
+  private Context createContext(Properties properties, Optional<CuratorFramework> zkClient) throws Exception {
+
+    Context.Builder contextBuilder = new Context.Builder()
+            .with(SHELL_VARIABLES, () -> variables)
+            .with(STELLAR_CONFIG, () -> properties);
+
+    // load global configuration from zookeeper
+    if (zkClient.isPresent()) {
+      Map<String, Object> global = fetchGlobalConfig(zkClient.get());
+      contextBuilder
+              .with(GLOBAL_CONFIG, () -> global)
+              .with(ZOOKEEPER_CLIENT, () -> zkClient.get())
+              .with(STELLAR_CONFIG, () -> getStellarConfig(global, properties));
+    }
+
+    return contextBuilder.build();
+  }
+
+  /**
+   * Fetches the global configuration from Zookeeper.
+   * @param zkClient The Zookeeper client.
+   * @return The global configuration retrieved from Zookeeper.
+   * @throws Exception
+   */
+  private Map<String, Object> fetchGlobalConfig(CuratorFramework zkClient) throws Exception {
+    byte[] raw = readGlobalConfigBytesFromZookeeper(zkClient);
+    return JSONUtils.INSTANCE.load(
+            new ByteArrayInputStream(raw),
+            new TypeReference<Map<String, Object>>() {});
+  }
+
+  /**
+   * Register all magic commands that will be available in the Stellar
+   * execution environment.
+   * @return A registry of magic commands.
+   */
+  private List<SpecialCommand> registerSpecialCommands() {
+    List<SpecialCommand> specials = Arrays.asList(
+            new AssignmentCommand(),
+            new DocCommand(),
+            new QuitCommand(),
+            new Comment(),
+            new MagicListFunctions(),
+            new MagicListVariables(),
+            new MagicDefineGlobal(),
+            new MagicUndefineGlobal(),
+            new MagicListGlobals()
+    );
```

Comment:
All the specials are made available here.

Reply:
If we ever do something like add assignment to the core Stellar language (like #687) then all we have to do is not use/remove the `AssignmentCommand` class.